lib/events/firestoreevents/firestoreevents_test.go

```diff
@@ -61,53 +59,69 @@ func (s *FirestoreeventsSuite) SetUpSuite(c *check.C) {
 	config.UIDGenerator = utils.NewFakeUID()
 
 	log, err := New(config)
+	require.NoError(t, err)
+
+	tt := &firestoreContext{
+		log: log,
+		suite: test.EventsSuite{
+			Log:        log,
+			Clock:      fakeClock,
+			QueryDelay: time.Second * 5,
+		},
+	}
 
-	c.Assert(err, check.IsNil)
-	s.log = log
-	s.EventsSuite.Log = log
-	s.EventsSuite.Clock = fakeClock
-	s.EventsSuite.QueryDelay = time.Second * 5
-}
+	t.Cleanup(func() {
+		tt.Close(t)
+	})
 
-func emulatorRunning() bool {
-	con, err := net.Dial("tcp", "localhost:8618")
-	if err != nil {
-		return false
-	}
-	con.Close()
-	return true
+	return tt
 }
 
-func (s *FirestoreeventsSuite) TearDownSuite(c *check.C) {
-	if s.log != nil {
-		s.log.Close()
-	}
-}
+func (tt *firestoreContext) Close(t *testing.T) {
+	ctx := context.Background()
 
-func (s *FirestoreeventsSuite) TearDownTest(c *check.C) {
 	// Delete all documents.
-	ctx := context.Background()
-	docSnaps, err := s.log.svc.Collection(s.log.CollectionName).Documents(ctx).GetAll()
-	c.Assert(err, check.IsNil)
+	docSnaps, err := tt.log.svc.Collection(tt.log.CollectionName).Documents(ctx).GetAll()
+	require.NoError(t, err)
 	if len(docSnaps) == 0 {
 		return
 	}
-	batch := s.log.svc.Batch()
+	batch := tt.log.svc.Batch()
 	for _, docSnap := range docSnaps {
 		batch.Delete(docSnap.Ref)
 	}
 	_, err = batch.Commit(ctx)
-	c.Assert(err, check.IsNil)
+	require.NoError(t, err)
+
+	if tt.log != nil {
+		err := tt.log.Close()
+		require.NoError(t, err)
+	}
 }
 
-func (s *FirestoreeventsSuite) TestSessionEventsCRUD(c *check.C) {
-	s.SessionEventsCRUD(c)
+func TestSessionEventsCRUD(t *testing.T) {
+	tt := setupFirestoreContext(t)
```

Comment:
Previously, firestore context was set up once per suite, now it's setup in each test. Is it intensive operation? I think we've been recently moving towards the opposite - i.e. for app/db tests we now set up all infra once and run all tests against it (at least, for integration tests). Maybe do the same here?

Reply:
Refactored so it works more like the old tests (broke out setting up of the suite then setting up of a test). Not sure if this reads any better to be honest.